scripts/pi-hole/php/teleporter.php

```diff
@@ -300,171 +300,121 @@ function archive_add_directory($path, $subdir = '')
     }
 }
 
-function limit_length(&$item, $key)
-{
-    // limit max length for a domain entry to 253 chars
-    // return only a part of the string if it is longer
-    $item = substr($item, 0, 253);
-}
-
-function process_file($contents)
-{
-    $domains = array_filter(explode("\n", $contents));
-    // Walk array and apply a max string length
-    // function to every member of the array of domains
-    array_walk($domains, 'limit_length');
-
-    return $domains;
-}
-
-function noun($num)
+function load_teleport_archive($filename, $flushtables = true, $blacklist = true, $whitelist = true, $regex_blacklist = true, $regex_whitelist = true, $regexlist = true, $auditlog = true, $adlist = true, $group = true, $client = true, $staticdhcpleases = true, $localdnsrecords = true, $localcnamerecords = true)
```

Comment:
Why are you passing 14 parameters to this function?

```suggestion
function load_teleport_archive($filename)
```

All other parameters are using values from the `$_POST` variable.
You can access `$_POST` inside the function. This is a global variable and it's available in all scopes.

There is no need to change the other lines.

Reply:
You already changed all old `$_POST['xxxxx']` to variables.

When you remove the parameters, you just need to add something like this inside the function:

```php
    $flushtables = isset($_POST['flushtables']); 
    $blacklist = isset($_POST['blacklist']); 
    $whitelist = isset($_POST['whitelist']);
    ...
```

Reply:
Whether passed as a global value (i.e. POST) or as arguments, there are still 14 values used by this function.

If things are changed to read POST instead, should the values be inserted ito POST around line 580? If not, how should these values be determined when called from the command line?

Reply:
No... no need to change `$_POST`;

I still think passing them as parameters is unnecessary.
You are passing these parameters only to have default values to each one.

You can add this inside the function:

```php
    $flushtables = isset($_POST['flushtables']) ? $_POST['flushtables'] : true;
    $blacklist = isset($_POST['blacklist']) ? $_POST['blacklist'] : true;
    $whitelist = isset($_POST['whitelist']) ? $_POST['whitelist'] : true;
    ...
    $variable = isset($_POST['something']) ? $_POST['something'] : "default value";
```